test/end2end_test.go

```diff
@@ -3901,6 +3901,36 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+func testClientInvalidStreamID(t *testing.T, e env) {
+	te := newTest(t, e)
+	ts := &funcServer{streamingInputCall: func(stream testgrpc.TestService_StreamingInputCallServer) error {
+		_, err := stream.Recv()
+		return err
+	}}
+	te.startServer(ts)
```

Comment:
`funcServer` is not required for this case. You can pass in `nil` to `startServer`.  (this can be fully avoided if you use a gRPC server instead of a test server)

Reply:
I didn't use a gRPC server as the servertester takes care of setting up both client and server for you.